aiohttp/helpers.py

```diff
@@ -879,3 +879,22 @@ def populate_with_cookies(
     for cookie in cookies.values():
         value = cookie.output(header="")[1:]
         headers.add(hdrs.SET_COOKIE, value)
+
+
+# https://tools.ietf.org/html/rfc7232#section-2.3
+_ETAGC = "[{}]+".format(
+    "".join(
+        chr(c) for c in (0x21,) + tuple(range(0x23, 0x7E)) + tuple(range(0x80, 0xFF))
+    )
+)
```

Comment:
You could use character ranges:
```suggestion
_ETAGC = r"[!#-}\x80-\xff]+"
```
Or even better:
```suggestion
_ETAGC = r"[^\x00- "\x7f]+"
```

Reply:
I prefer the first suggestion: don't want to fight with regexps when non-ASCII symbol appears in a tested string somehow

Reply:
What encoding is used for decoding headers? How bytes `\x80-\xff` are mapped?

BTW there are needed tests for non-ASCII ETags.

Reply:
Non-ASCII cases successfully added.